docs/docs/reference/other-new-features/indentation-experimental.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: doc-page
+title: "Fewer Braces"
+---
+
+By and large, the possible indentation regions coincide with those regions where braces `{...}` are also legal, no matter whether the braces enclose an expression or a set of definitions. There is one exception, though: Arguments to function can be enclosed in braces but they cannot be simply indented instead. Making indentation always significant for function arguments would be too restrictive and fragile.
+
+To allow such arguments to be written without braces, a variant of the indentation scheme is implemented under language import
+```scala
+import language.experimental.fewerBraces
+```
+Alternatively, it can be enabled with command line option `-language:experimental.fewerBraces`.
+
+This variant is more contentious and less stable than the rest of the significant indentation scheme. It allows to replace a function argument in braces by a `:` at the end of a line and indented code, similar to the convention for class bodies. It also allows to leave out braces around arguments that are multi-line function values.
+
+## Using `:` At End Of Line
+
+
+Similar to what is done for classes and objects, a `:` that follows a function reference at the end of a line means braces can be omitted for function arguments. Example:
+```scala
+times(10):
+   println("ah")
+   println("ha")
+```
+
+Function calls that take multiple argument lists can also be handled this way:
+
+```scala
+val firstLine = files.get(fileName).fold:
+      val fileNames = files.values
+      s"""no file named $fileName found among
+         |${values.mkString(\n)}""".stripMargin
+   :
```

Comment:
Are we able to handle this another way?  It seems a bit wrong? Kind of like someone left the colon by mistake.

Reply:
Maybe something like a soft keyword `and` ?

Reply:
I think curried functions with complex arguments are pretty much an anti-pattern anyway, but others like their `Option.fold`s. The current possibility falls out from the rules we have. Before this PR there was a special added rule that disallowed the `:` at start of line. But that would leave no way to handle curried functions except with braces. So I think it's probably better to have simpler, more orthogonal rules and let users decide whether they like this pattern or whether they want to keep using braces after all.

Reply:
It feels like it might look better if there will be an additional per-parameter indentation.
Smth like:
```scala
val firstLine = files.get(filename).fold:
  {smth here} =>
     val filenames = files.values
     s"""no file named $fileName found among
         |${values.mkString(\n)}""".stripMargin
  f =>
     val lines = f.iterator.map(_.readline)
     lines.mkString("\n")
```

Reply:
That does not work. This would group both lambdas in a sequence as a single parameter block.

Reply:
Yep, I missed that `:` is about different parameter sections.

Should this work if `:` have the same indentation level as the first parameter?

```scala
val firstLine = files.get(filename).fold:
  val filenames = files.values
  s"""no file named $fileName found among 
      |${values.mkString(\n)}""".stripMargin
  :
  f =>
    val lines = f.iterator.map(_.readline)
    lines.mkString("\n")
```

Reply:
No, then it does not work. Everything at the same indentation level is one argument